sky/adaptors/aws.py

```diff
@@ -98,21 +98,31 @@ def resource(service_name: str, **kwargs):
 
     Args:
         service_name: AWS resource name (e.g., 's3').
-        kwargs: Other options.
+        kwargs: Other options. We add max_attempts to the kwargs instead of
+            using botocore.config.Config() because the latter is not hashable.
     """
+    assert 'config' not in kwargs, (
+        'config should not be passed to resource(), '
+        'as botocore.config.Config() is not hashiable.')
     if not hasattr(_local, 'resource'):
+        # TODO(zhwu): this dict should be LRU cache, instead of growing forever,
+        # to avoid issues in #2668.
         _local.resource = {}
 
     # Using service name and kwargs as key
     sorted_kwargs = tuple(sorted(kwargs.items(), key=lambda x: x[0]))
     key = (service_name, sorted_kwargs)
+    max_attempts = kwargs.pop('max_attempts', None)
```

Comment:
Should we add the same protection to client() below? It also uses a cache.

Reply:
Added the assertion for all the places that uses the cache now.